src/handlers/initiateDataRequest.ts

```diff
@@ -1,18 +1,28 @@
 import { APIGatewayProxyResult, APIGatewayProxyEvent } from 'aws-lambda'
 import { initiateDataTransfer } from '../services/initiateDataTransfer'
 import { updateZendeskTicket } from '../services/updateZendeskTicket'
+import { isValidSignature } from '../services/validateRequestSource'
 import { validateZendeskRequest } from '../services/validateZendeskRequest'
 import { DataRequestParams } from '../types/dataRequestParams'
 import { ValidatedDataRequestParamsResult } from '../types/validatedDataRequestParamsResult'
 
 export const handler = async (
   event: APIGatewayProxyEvent
 ): Promise<APIGatewayProxyResult> => {
+  const headerSignature = event.headers['X-Zendesk-Webhook-Signature'] as string
+  const timestamp = event.headers[
+    'X-Zendesk-Webhook-Signature-Timestamp'
+  ] as string
+  if (
+    !(await isValidSignature(headerSignature, event.body as string, timestamp))
```

Comment:
If this is the only place we use the function it might be better to call it `isSignatureInvalid` and return true if it's not valid.  That way we don't need the not operator.

Reply:
I'm also not sure about casting the body and headers to strings - could they be undefined, is that something we should handle in the function where we validate the signature?

Reply:
addressed in latest commit. Please review